drivers/pinmux/pinmux_b91.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2021 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT telink_b91_pinmux
+
+#include "analog.h"
+#include <drivers/pinmux.h>
+
+
+/**
+ *      GPIO Function Enable Register
+ *         ADDR              PINS
+ *      gpio_en:          PORT_A[0-7]
+ *      gpio_en + 1*8:    PORT_B[0-7]
+ *      gpio_en + 2*8:    PORT_C[0-7]
+ *      gpio_en + 3*8:    PORT_D[0-7]
+ *      gpio_en + 4*8:    PORT_E[0-7]
+ *      gpio_en + 5*8:    PORT_F[0-7]
+ */
+#define reg_gpio_en(pin) (*(volatile uint8_t *)((uint32_t)DT_INST_REG_ADDR_BY_NAME(0, gpio_en) + \
+						((pin >> 8) << 3)))
```

Comment:
Nit: Unless `<< 3` is semantically a bitfield operation, `* 8` should be preferred as that makes it clearer as to what is being done.
```suggestion
#define reg_gpio_en(pin) (*(volatile uint8_t *)((uint32_t)DT_INST_REG_ADDR_BY_NAME(0, gpio_en) + \
						((pin >> 8) * 8)))
```

Reply:
Fixed in all places.